File: marcel-stdlib/src/main/java/marcel/lang/printer/Printer.java
package marcel.lang.printer;

public interface Printer {

  void print(CharSequence s);
  default void print(Object o) {
    print(o instanceof CharSequence ? (CharSequence) o : String.valueOf(o));
  }
  void println(CharSequence s);
  default void println(Object o) {
    println(o instanceof CharSequence ? (CharSequence) o : String.valueOf(o));
  }
  void println();

}

File: subprojects/marcel-repl/src/main/kotlin/com/tambapps/marcel/repl/printer/Printer.kt
package com.tambapps.marcel.repl.printer

/**
 * Interface representing an object printing data
 */
interface Printer {
  suspend fun print(s: CharSequence?)
  suspend fun print(o: Any?) {
    print(if (o is CharSequence) o else o.toString())
  }

  suspend fun println(s: CharSequence?)
  suspend fun println(o: Any?) {
    println(if (o is CharSequence) o else o.toString())
  }

  suspend fun println()
}